### PR TITLE
[bc test] pull nightly from before the base commit

### LINKED ISSUE #79146: Pin PyTorch nightly used for comparison in BC test
### 🐛 Describe the bug

Today the backwards compatibility test is a common source of spurious failures on pull request because if your pull request is too old then it will not contain all of the operators from the latest nightly and the test will incorrectly assumed that you remove them. Working with our commit hash pinning, I realized that now that we are GH1 we can do something similar for BC test: we will maintain a file that points to the nightly that should be used for the backwards compatibility test and whenever the new nightly is finished building we run a periodic can GitHub actions job to commit to master with an update. WDYT.

cc @seemethere @malfet @pytorch/pytorch-dev-infra 

### Versions

master

### PR DESCRIPTION
**Pull request by janeyx99**
Fixes #79146

Test plan:
https://github.com/pytorch/pytorch/runs/6902235940?check_suite_focus=true passes and installs yesterday's nightly

new test also passes: https://github.com/pytorch/pytorch/runs/6903723789?check_suite_focus=true#step:9:640

### CODE REVIEW

File: .jenkins/pytorch/test.sh [hunk 1]
```diff
@@ -457,7 +457,8 @@ test_forward_backward_compatibility() {
   python -m venv venv
   # shellcheck disable=SC1091
   . venv/bin/activate
-  pip_install --pre torch -f https://download.pytorch.org/whl/nightly/cpu/torch_nightly.html
+  DATE_OF_BASE=$(git show -s --format=%cd --date=short ${BASE_SHA})
+  pip_install --pre "torch<1.13.0.dev${DATE_OF_BASE//-/}" -f https://download.pytorch.org/whl/nightly/cpu/torch_nightly.html
```
Comment (janeyx99) [inline .jenkins/pytorch/test.sh:461]: Feeling iffy about hardcoding the 1.13.0.dev here --> are there any better ways of specifying a nightly prior to a certain date OR dynamically generating the 1.13.0.dev portion?

Comment (albanD) [reply .jenkins/pytorch/test.sh:461]: You do have the current pytorch installed before activative this venv. you can get the head's version to generate the 1.13.0 part.

Comment (malfet) [reply .jenkins/pytorch/test.sh:461]: One can read a version from [version.txt](https://github.com/pytorch/pytorch/blob/master/version.txt), can't they?

File: .github/workflows/_linux-test.yml [hunk 2]
```diff
@@ -72,6 +72,7 @@ jobs:
           PR_NUMBER: ${{ github.event.pull_request.number }}
           BRANCH: ${{ steps.parse-ref.outputs.branch }}
           SHA1: ${{ github.event.pull_request.head.sha || github.sha }}
+          BASE_SHA: ${{ github.event.pull_request.base.sha || github.sha }}
```
[the remark this answers is not in the document]
Comment (janeyx99) [reply .github/workflows/_linux-test.yml:75]: The commit the workflow is currently running on (so this would be in the case of master, I suppose)

**Review by malfet — APPROVED** [review]
LGTM, but please consider using version from version.txt (something like `X=$(cat version.txt); echo ${X:0:-2}`, @seemethere bash-foo is much better than mine though)

File: .jenkins/pytorch/test.sh (hunk 1 above)
Comment (albanD) [reply .jenkins/pytorch/test.sh:461]: I never know if this is generated/updated automatically in some way during the build process?

**Comment by malfet** [comment]
Just realized, that none of the approaches would work on the day of the release version update. Perhaps we should just grep for a date from https://download.pytorch.org/whl/nightly/cpu/torch ?

**Comment by janeyx99** [comment]
> Just realized, that none of the approaches would work on the day of the release version update. Perhaps we should just grep for a date from https://download.pytorch.org/whl/nightly/cpu/torch ?

I didn't want to grep a specific date because then, if the nightlies didn't run/upload on that date, the check would also fail. (This was why I used < instead of == in the versioning.) 

Regardless, I will add a fallback where if the specific install doesn't work, we will just go back to installing the latest nightly, which seems like a fine thing to do for our initial use version.txt approach too?

**Review by albanD — COMMENTED** [review]
SGTM lets look how it fares in practice!

**Review by albanD — APPROVED** [review]

**Comment by janeyx99** [comment]
@pytorchbot merge -f

Not going to wait for all checks as bc already passes

**Pull request closed without merging**